app/models/appointment.rb

```diff
@@ -65,6 +65,10 @@ def self.eligible_for_reminders(days_overdue: 3)
     overdue_by(days_overdue)
       .joins(:patient)
       .merge(Patient.contactable)
+      .includes(:communications)
```

Comment:
As you may recall, there's a bug in how notifications currently work. We send the same messages daily because this query looks for appointments that are 3+ days overdue. Then it skips if the appointment has a successful twilio response. Which means any failed communications will attempt to send every day. 

This changes the behavior to only include appointments if they don't have either a communication or appointment reminder. That's an improvement because we shouldn't keep trying if the communication failed. It might occasionally be successful the second time, but more often it will just fail again, either because the person doesn't have whatsapp or the number is no longer valid. Any notification that has ever been attempted (and didn't fail due to a sidekiq or twilio client error) should have a communication.

We also exclude any appointments that have reminders because an appointment that has a reminder but no communication is part of an experiment and will be handled by the experiment cron.

Reply:
should be explicit `left_joins`

Reply:
Thanks for catching this.